### DISCUSSION TITLE
Chewtils 2.0

### DISCUSSION THREAD

**Discussion by Chew** [Chew/JDA-Chewtils#88] [Announcements]
After years of prep-work, JDA 5 waiting, breaking changes, rory, and more, it is finally time to release 2.0.

Chewtils 2.0 has full support for JDA 5. And, despite all the changes, is still _mostly_ backwards compatible with JDA-Utilities. The only breaking change is `CooldownScope` being moved to its own class.

If you're just hopping into JDA-Chewtils, welcome aboard! Your experience over will be the smoothest. Outside of needing to import `CooldownScope`, you are free to start using new methods.

As for everyone else, there are now breaking changes to be made aware of. All breaking changes from v1 to v2 are documented [on the wiki](https://github.com/Chew/JDA-Chewtils/wiki/Breaking-Changes-in-JDA-Chewtils-2.0).

This release is the exact same as the latest 2.0-SNAPSHOT release, so there's no changes since then to report.

Thank you all for reporting bugs throughout JDA's Beta 5.

## What's Changed (since 1.24.1)
* Update to JDA 5 & bump to JDA-Chewtils 2.0 by @rtm516 in https://github.com/Chew/JDA-Chewtils/pull/35
* Support for the interactions rework & Context Menus by @Chew in https://github.com/Chew/JDA-Chewtils/pull/45
* Fix Context Menus not being found by @Matyrobbrt in https://github.com/Chew/JDA-Chewtils/pull/49
* Add SlashCommandEvent#optAttachment() by @Chew in https://github.com/Chew/JDA-Chewtils/pull/54
* Utilize new getOption methods from JDA in optX by @Chew in https://github.com/Chew/JDA-Chewtils/pull/55
* Add a ButtonEmbedPaginator with support for Discord buttons by @Chew in https://github.com/Chew/JDA-Chewtils/pull/50
* Full Support for JDA 5.0.0-alpha.13 (Permissions/Emoji) by @Chew in https://github.com/Chew/JDA-Chewtils/pull/67
* Correct mistake in JavaDocs (SlashCommand) by @olliedean in https://github.com/Chew/JDA-Chewtils/pull/69
* Bump dependencies (slf4j, okhttp, org.json) by @JellyBrick in https://github.com/Chew/JDA-Chewtils/pull/70
* Support JDA 5.0.0-alpha.15 by @Chew in https://github.com/Chew/JDA-Chewtils/pull/72
* Use better method for finding the proper slash command by @Chew in https://github.com/Chew/JDA-Chewtils/pull/63
* Support for JDA 5.0.0-alpha.18 (Message Refactor) by @Andre601 in https://github.com/Chew/JDA-Chewtils/pull/76
* Support for JDA 5.0.0-alpha.19 by @OurTwinkie in https://github.com/Chew/JDA-Chewtils/pull/77
* Add Interaction Localization support by @OurTwinkie in https://github.com/Chew/JDA-Chewtils/pull/78
* Support for v5.0.0-alpha.21 by @book000 in https://github.com/Chew/JDA-Chewtils/pull/79
* Fix slash command children by @ipsvn in https://github.com/Chew/JDA-Chewtils/pull/80
* Fix oauth 2 breaking on JDA Beta7 onwards by @astra-OCE in https://github.com/Chew/JDA-Chewtils/pull/83
* Fix slash command children in groups matching children with the same name by @Matyrobbrt in https://github.com/Chew/JDA-Chewtils/pull/81
* Bugfix ButtonEmbedPaginator finalAction by @TheMather1 in https://github.com/Chew/JDA-Chewtils/pull/85
* add a translate util file by @Chew in https://github.com/Chew/JDA-Chewtils/pull/87

## New Contributors
* @Matyrobbrt made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/49
* @olliedean made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/69
* @JellyBrick made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/70
* @OurTwinkie made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/77
* @book000 made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/79
* @ipsvn made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/80
* @astra-OCE made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/83
* @TheMather1 made their first contribution in https://github.com/Chew/JDA-Chewtils/pull/85

**Full Changelog**: https://github.com/Chew/JDA-Chewtils/compare/1.24.1...2.0

<hr /><em>This discussion was created from the release <a href='https://github.com/Chew/JDA-Chewtils/releases/tag/2.0'>Chewtils 2.0</a>.</em>

**Comment by Andre601** [comment]
Are there any chances to get a localization file support for other file types (i.e. json) in the future?
Or a way to implement your own util here for chewtils to use?

I'm no big fan of properties files, especially given they don't support nesting/grouping values... So some way to retrieve values from other file-types would be a nice-to-have feature here.

**Reply by Chew** [reply]
Definitely could be considered, I certainly have no plans for it myself. I'll definitely accept PRs for it. It's currently just a simple thing I made and tossed into the library. The main thing is as long as it doesn't get too complicated for the end user. The main thing is those 2 methods I made for it, designed for the this.name/descriptionLocalization

**Reply by Andre601** [reply]
One small issue I kinda see here right now, is the fact that there isn't a page explaining this translation system properly (i.e. how you use it to provide translations for command name and description).

Because right now could I life with the fact of making a properties file for the command names and descriptions, while using my own util for everything else (Until I find a way to squeeze in a JSON file here for this stuff.), but without a clear example of how I would apply a translation to a command name and description with chewtils is this not easy for me to understand...
